store/postgres/src/detail.rs

```diff
@@ -176,78 +177,82 @@ impl TryFrom<ErrorDetail> for SubgraphError {
-            latest_ethereum_block_hash,
-            latest_ethereum_block_number,
-        )?;
-        let health = health.into();
-        let chain = status::ChainInfo {
-            network: site.network.clone(),
-            chain_head_block,
-            earliest_block,
-            latest_block,
-        };
-        let entity_count = entity_count.to_u64().ok_or_else(|| {
-            constraint_violation!(
-                "the entityCount for {} is not representable as a u64",
-                deployment
-            )
-        })?;
-        let fatal_error = error.map(SubgraphError::try_from).transpose()?;
-        // 'node' needs to be filled in later from a different shard
-        Ok(status::Info {
-            id: id.into(),
-            subgraph: deployment,
-            synced,
-            health,
-            fatal_error,
-            non_fatal_errors: vec![],
-            chains: vec![chain],
-            entity_count,
-            node: None,
-        })
-    }
+pub(crate) fn info_from_details(
+    detail: DeploymentDetail,
+    fatal: Option<ErrorDetail>,
+    non_fatal: Vec<ErrorDetail>,
+    sites: &[Arc<Site>],
+) -> Result<status::Info, StoreError> {
+    let DeploymentDetail {
+        id,
+        deployment,
+        failed: _,
+        health,
+        synced,
+        fatal_error: _,
+        non_fatal_errors: _,
+        earliest_ethereum_block_hash,
+        earliest_ethereum_block_number,
+        latest_ethereum_block_hash,
+        latest_ethereum_block_number,
+        entity_count,
+        graft_base: _,
+        graft_block_hash: _,
+        graft_block_number: _,
+        ..
+    } = detail;
+
+    let site = sites
+        .iter()
+        .find(|site| site.deployment.as_str() == deployment)
+        .ok_or_else(|| constraint_violation!("missing site for subgraph `{}`", deployment))?;
+
+    // This needs to be filled in later since it lives in a
+    // different shard
+    let chain_head_block = None;
+    let earliest_block = block(
+        &deployment,
+        "earliest_ethereum_block",
+        earliest_ethereum_block_hash,
+        earliest_ethereum_block_number,
+    )?;
+    let latest_block = block(
+        &deployment,
+        "latest_ethereum_block",
+        latest_ethereum_block_hash,
+        latest_ethereum_block_number,
+    )?;
+    let health = health.into();
+    let chain = status::ChainInfo {
+        network: site.network.clone(),
+        chain_head_block,
+        earliest_block,
+        latest_block,
+    };
+    let entity_count = entity_count.to_u64().ok_or_else(|| {
+        constraint_violation!(
+            "the entityCount for {} is not representable as a u64",
+            deployment
+        )
+    })?;
+    let fatal_error = fatal.map(SubgraphError::try_from).transpose()?;
+    let non_fatal_errors = non_fatal
+        .into_iter()
+        .map(SubgraphError::try_from)
+        .collect::<Result<Vec<SubgraphError>, StoreError>>()?;
```

Comment:
Git generated a huge diff here, but this is the only actual change.

Reply:
Hiding whitespace in the diff fixes it, but thanks for the heads up.

Reply:
Maybe it's time to finally try `diff.algorithm patience` :grimacing: